docs: initial upload of styles used by doc linter

bob-watson opened the pull request:
Adds styles and configuration files for contributors to use the Vale documentation linter in their IDE while they edit angular.io topics.
This commit includes documentation for using the documentation linter in the README.md file.
This commit does not alter any angular.io or other published documentation.

Commit:
docs: initial upload of styles used by doc linter

Commit:
docs: initial upload of styles used by doc linter

Commit:
docs: initial upload of styles used by doc linter

bob-watson commented:
This commit is going to cause problems with the inclusive language CI test because it includes files with all the prohibited words in its tests to look for those words in the documentation. 

Is there a way to mark those file(s) as allowable exceptions to this CI test? It looks like it's only two files:
* aio/tools/doc-linter/dictionaries/en_us/en_US-web.dic
* aio/tools/doc-linter/styles/Angular/Angular_Inclusive.yml

josmar-crwdstffng reviewed (approved):
@LGTM
Reviewed-for: global-docs-approvers

I have confirmed that the definitions work as intended and match the standalone repo.

Commit:
docs: edits from reviewers

dario-piotrowicz committed:
fix(animations): make sure falsy values are added to _globalTimelineStyles (#46863)

style values get added to the `_globalTimelineStyles` map in order to keep
them so that they can be used across different timelines

`_globalTimelineStyles` was previously a plain object but has been
refactored to a map in #44482, as part of the update a check has been
changed from a ternary operation to an or (||), causing falsy values (as 0)
not to be added to the map anymore, apply the nullish coalescing operator (??)
instead to make sure only `undefined` and `null` are filtered out

also since this aspect was clearly not covered by tests, add a new test
to ensure that such regression doesn't happen in the future

resolves #46833

PR Close #46863

gkalpak commented on aio/tools/doc-linter/README.md:
```diff
@@ -0,0 +1,191 @@
+Vale evaluates a markdown (`.md`) file when you open it in the IDE and each time you save it.
+
+Fixing the problems that Vale identifies can range from simple word substitutions to rewriting sentences or paragraphs.
+Keep in mind that the goal is to make your documentation  as clear and easy to read as you can.
```
```suggestion
Keep in mind that the goal is to make your documentation as clear and easy to read as you can.
```

bob-watson replied:
Removed this in the current edit.

gkalpak commented on aio/tools/doc-linter/README.md:
```diff
@@ -0,0 +1,191 @@
+**Before you proceed make sure that:**
+
+1. You have installed the [`angular/angular` repo](https://github.com/angular/angular).
+1. You have a development or writing environment for `angular.io` in the `angular/angular` repo and you can build the `angular.io` docs. For information about creating the authoring environment for `angular.io`, see the [Angular documentation project](https://github.com/bob-watson/angular/tree/doc-linter/aio#angular-documentation-project-httpsangulario)
```
Inconsistent numbering. Most other numbered lists use increasing numbers.

bob-watson replied:
updated

gkalpak commented on aio/tools/doc-linter/README.md:
```diff
@@ -0,0 +1,191 @@
+
+**On a Mac, locally run these tools:**
+
+1. Open Chrome.
+1. Open a command-line tool, such as `terminal`.
+1. Open VS Code.
```
Inconsistent numbering. Most other numbered lists use increasing numbers.

```suggestion
1. Open Chrome.
2. Open a command-line tool, such as `terminal`.
3. Open VS Code.
```

gkalpak commented on aio/tools/doc-linter/README.md:
```diff
@@ -0,0 +1,191 @@
+
+**Before you proceed make sure that:**
+
+1. You have installed the [`angular/angular` repo](https://github.com/angular/angular).
```
The term "installed" is a little ambiguous here. You can't really install a git repo.
Did you mean (a) [clone the `angular/angular` repo][1], (b) clone [your fork][2] of the `angular/angular` repo or (c) having cloned the repo (or fork) to run `yarn install` (to install dependencies)?

[1]: https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository
[2]: https://docs.github.com/en/get-started/quickstart/fork-a-repo
[3]:

gkalpak commented on aio/tools/doc-linter/README.md:
```diff
@@ -0,0 +1,191 @@
+1. Install the `Vale` VS Code extension.
+     1. Search for `Vale`.
+     2. Install `Vale`.
+2. Navigate to **Preferences > Settings**. The **Preferences** menu item is located under the **Code** nav bar on macOS.
```
Inconsistent casing of `macOS`. Most other occurrences in this document use `MacOS` (although I believe `macOS` is more correct).

bob-watson replied:
fixed.

gkalpak commented on aio/tools/doc-linter/README.md:
```diff
@@ -0,0 +1,191 @@
+   3. If this command does not return a path, restart your Mac and repeat this step.
+3. Update the instance of the `angular` repo on your system.
+4. In the command line tool:
+   1. Navigate to your `github` workspace directory.
```
Did you mean "git workspace" or "git working directory" (here and below)?
I don't think "github workspace directory" is a thing (but I might be wrong).

bob-watson replied:
It should be (and now is) *workspace directory*. Thanks!

gkalpak commented on aio/tools/doc-linter/README.md:
```diff
@@ -0,0 +1,191 @@
+   1. Navigate to your `github` workspace directory.
+   2. In your `github` workspace, navigate to the `angular/aio/tools/doc-linter` directory.
+   3. In the `angular/aio/tools/doc-linter` directory, confirm that you have the `vale.ini` file.
+   4. Run `pwd` to get the path to the `vale.ini` file for use the next procedure.
```
```suggestion
   4. Run `pwd` to get the path to the `vale.ini` file for use in the next procedure.
```

bob-watson replied:
fixed

atscott committed:
refactor(router): Update TitleStrategy to useFactory (#46876)

The implementation of the `DefaultTitleStrategy` was modeled after the
existing strategy patterns in the Router. These patterns were developed
before the `providedIn` syntax for injectables. We can simplify the
model a lot by providing the default in the factory of the abstract
class.

Note that the other strategy patterns aren't touched in this PR due to
how long they've existed. Because they have been there for such a long
time, it's possible there will need to be some adjustments to code
if/when they are refactored to do the same.

PR Close #46876

bob-watson commented on aio/tools/doc-linter/styles/Angular/Angular_Code-as-word.yml:
```diff
@@ -0,0 +1,8 @@
+# tests for strings qualified by back-ticks are surrounded by spaces or punctuation
```
Thanks for catching this!

bob-watson commented on aio/tools/doc-linter/README.md:
```diff
@@ -0,0 +1,191 @@
+1. [Install Vale](#install-vale-in-your-vs-code-ide) in your VS Code IDE or, if you use a different IDE, see the [Vale documentation](https://vale.sh/docs/vale-cli/installation/) for information about installing and adding it.
+2. View the problems in your IDE and fix them.
+ 
+Log any problem you encounter with the documentation linter as [a new docs issue](https://github.com/angular/angular/issues/new?assignees=&labels=&template=3-docs-bug.yaml) in the repo. Give your new issue the `comp: docs` and `docsarea: linter` labels.
```
Thanks. I rewrote this to be appropriate to a wider audience.

bob-watson commented on aio/tools/doc-linter/README.md:
```diff
@@ -0,0 +1,191 @@
+**Before you proceed make sure that:**
+
+1. You have installed the [`angular/angular` repo](https://github.com/angular/angular).
+1. You have a development or writing environment for `angular.io` in the `angular/angular` repo and you can build the `angular.io` docs. For information about creating the authoring environment for `angular.io`, see the [Angular documentation project](https://github.com/bob-watson/angular/tree/doc-linter/aio#angular-documentation-project-httpsangulario)
```
I did. Thanks!

bob-watson commented on aio/tools/doc-linter/vale.ini:
```diff
@@ -0,0 +1,17 @@
+
+[*.md]
+
+BlockIgnores = (?m) *(<code-example[^>]*>[\s\S\n]*<\/code-example>)
```
See previous comment.

atscott committed:
fix(core): Fix `runInContext` for `NgModuleRef` injector (#46877)

The `runInContext` for `NgModuleRef` was previously an infinite loop.

PR Close #46877

angular-robot committed:
build: update babel dependencies to v7.18.9 (#46883)

See associated pull request for more information.

PR Close #46883

angular-robot committed:
build: update eslint dependencies to v5.30.7 (#46884)

See associated pull request for more information.

PR Close #46884

angular-robot committed:
build: update github/codeql-action action to v2.1.16 (#46885)

See associated pull request for more information.

PR Close #46885

devversion committed:
build: update source-map package to latest version (#46888)

The source-map package is updated as part #46707, but the minor
version bump introduced breaking changes. This commit extracts the
update so that we can handle it separately accounting for the breaking
changes. Also removing the caret since this package apparently makes
breaking changes in minor segment bumps.

PR Close #46888

devversion committed:
build: remove `source-map` package from `compiler-cli` bundling external list (#46888)

The source-map package is no longer explicitly used in the
`compiler-cli` package and therefore can be removed from the ESBuild
bundling, denoting it as external. This should be a noop.

PR Close #46888

devversion committed:
test: update source-map tests to account for `source-map` breaking change (#46888)

The source-map package now requires the
`SourceMapConsumer`/`SourceMapGenerator` classes to be instantiated
asynchronously. This commit updates our tests to account for that.

PR Close #46888

devversion committed:
build: disable jasminewd2 types for sourcemap playground test (#46888)

The sourcemap test in the e2e playground is now using async/await code.
This results in errors now with the Bazel TS compilation because it
detects that `expect` is returning a promise and should be awaited.

This happens due to the jasminewd2 types. We should just use the actual
jasmine types and not rely on the deprecated selenium control flow,
using explicit async/await in the whole test. This also solves the issue
with the source-map types being async/await now.

PR Close #46888

devversion committed:
test: convert `source-map` core tests to use `async/await` instead of `fakeAsync` (#46888)

The source map tests rely on asynchronous logic from the `source-map`
package. The tests itself are written using `fakeAsync` but this
unnecessarily complicates the interaction with the asynchronous
source-map helpers/package.

To fix this, we just make the tests use async/await as we don't intend
to test fakeAsync in this `describe` block..

PR Close #46888

bob-watson commented on aio/tools/doc-linter/styles/Angular/Angular_Note.yml:
```diff
@@ -0,0 +1,11 @@
+ignorecase: false
+scope: raw
+raw:
+  - (?<!\*\*)NOTE[^\*]*(?!\*\*)
+  - (?<!\*\*)TIP[^\*]*(?!\*\*)
+  - (?<!\*\*)IMPORTANT[^\*]*(?!\*\*)
```
I'd like to tighten them up under a separate issue/PR if that would work for you.
These currently work(-ish) for our content, but I agree that they could be improved.

The intention is to find cases of `NOTE|TIP|IMPORTANT` that are:
* all caps
* the only word on the line
* not wrapped by bold notation `**` on both sides

Commit:
Merge branch 'doc-linter-review-edit' into doc-linter

Commit:
Merge branch 'doc-linter' of https://github.com/bob-watson/angular into doc-linter

bob-watson commented:
The commits in this branch got all tangled up and look worse than the actual changes that were made.
Closing this PR and replacing with PR #46897.
